service/src/main/resources/api/openapi.yml

```diff
@@ -35,9 +35,52 @@ paths:
               schema:
                 $ref: '#/components/schemas/VersionProperties'
 
+  # User profile management
+  /api/profile:
+    parameters:
+      - $ref: '#/components/parameters/ProfilePath'
+    get:
+      summary: Retrieve a profile property.
```

Comment:
Nice to add details (summary or description), for example for someone calling this API from swagger. If ProfilePath is `/`, will they get entire user profile?

For setUserProfile, ProfilePath must have at least one "directory".

Also good to mention that entire user profile is returned, not just the part that was set. (Should there be a test for this?)

Also, setUserProfile is missing `parameters`, not sure how things are working..

Reply:
Yep fair enough, those descriptions could use some better documentation.

As for the others, ATM the put endpoint should be returning the value that was just passed in by calling getUserProfile on the same path. Parameters - they are specified at the top level so both the get and put should use them is my understanding?

Reply:
SG, I misread the code, thanks